sky/serve/examples/generate_middleware_yaml/task_in_middleware.yaml

```diff
@@ -0,0 +1,12 @@
+resources:
+  cloud: gcp
+  ports:
+  - 8081
+
+workdir: ~/sky_workdir
+
+run: python3 server.py
+
+service:
+  port: 8081
```

Comment:
Is it possible to make `resources.ports` optional and auto-filled when `service.port` is specified? That seems to be a UX improvement. Once people wrote the `service` dict the port will be auto-exposed. In the current way, it may be hard to remember to write 2 places.

Reply:
It is already optional; just add auto filling 🫡